src/source/internal.rs

```diff
@@ -316,10 +316,10 @@ impl Source for Internal {
                     }
                 }
             } else {
```

Comment:
Can you add a comment here?  The branch above is so large it isn't immediately obvious to which cases this else applies.

Reply:
Good idea.